front-end/src/store/modules/material.store.ts

```diff
@@ -0,0 +1,90 @@
+import { defineStore } from 'pinia';
+
+import { isEmpty } from '@/utils/string.helper';
+
+import { getMaterials, createMaterial, updateMaterial, deleteMaterial } from '@/api/material-req';
+import { withErrorHandler } from '@/utils/storeHandler';
+import { Material, MaterialStore } from '@/store/interfaces/material.interface';
+
+export const useMaterialStore = defineStore('materialStore', {
+	state: (): {
+		filter: { input: string; site: string; type: string; state: string };
+		input: string;
+		materials: Array<Material>;
+	} => {
+		return {
+			materials: [],
```

Comment:
That's fine, but then you can't reset your store on logout.
Just letting you known.